### DISCUSSION TITLE
How can I add new steps to the preprompts?

### DISCUSSION THREAD

**Discussion by tksix** [AntonOsika/gpt-engineer#1163] [Q&A]
Hi, is it only possible to customize the existing preprompt files or is it also possible to add completely new steps in the preprompts folder and add it to the process for example “test_generation”. If so, could someone provide a brief step-by-step guide on how to accomplish this? Many thanks in advance!

**Comment by gl2007** [comment]
I know this is an old thread, but I was going through the code today. There is a parameter use_custompreprompts which makes it look for preprompts within the "new project" folder. You can perhaps customize the prompts, add new prompts etc. Have not tried it myself but worth a try.
